engine/src/main/java/org/terasology/rendering/collada/ColladaLoader.java

```diff
@@ -285,8 +285,13 @@ protected void parseSkeletalMeshData(Element rootElement) throws ColladaParseExc
                         if ("JOINT".equals(vertexWeightsInput.semantic)) {
                             md5Weight.jointIndex = index;
                             vertexWeightsJointNameArray[vertexWeightsIndex] = jointNameArray[index];
-                            // logger.debug(String.valueOf(vertexWeightsVDataIndex) + ": " + "jointName=" + vertexWeightsJointNameArray[vertexWeightsIndex]);
+                            if (vertexWeightsJointNameArray[vertexWeightsIndex] == null){
+                                logger.debug("No jointNameArray in index: " + index);
+                            }
                         } else if ("WEIGHT".equals(vertexWeightsInput.semantic)) {
+                            if (weightsArray == null) {
+                                logger.debug("No weights in weightsArray");
```

Comment:
Here, you need to actually handle the null, rather than just logging it. If you log it then that won't stop the error two lines below from still being hit and getting a `NullPointerReference`.